### PR TITLE
TGP-377: Implement Create Record endpoint

### PR DESCRIPTION
**Pull request by IoannisLazaridis-oc**
## Description
Implemented new endpoint for Create Record

- [x] **Endpoint for creating a record**: 
`POST https://<host>/customs/traders/goods-profiles/{eori}/records` 
`POST http://localhost:{port}/{eori}/records`
- [x] **Validating incoming request body**: Added `validateCreateRecordRequest()` to ensure all mandatory parameters are present.
- [x] **Authorization and header validation**: The controller now uses `authAction` and `validateHeader` actions to check for proper authorization and valid headers.
- [x] **Forwarding request to the router**: After validation, the request is prepared and sent to the router using `createRecordRequestForRouter()`.
- [x] Resolved conflicts from main branch  after PR #45  
- [x] Implemented validation for mandatory fields for Create Record Request

Note: The `InvalidErrorResponse` now can handle multiple errors in `errors` field

---
**Ticket: [TGP-377](https://jira.tools.tax.service.gov.uk/browse/TGP-377)**

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala
Comment (AmarkumarJagtap) [inline app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala:36]: think we need to wait until 673 is merged in as we have removed datetimeservice now from error responses and refactored some of the code around it. 673 is nearly done is some test cases needs to be sorted, we can pair on this ticket and update the rest of changes.

File: app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala:64]: I would put this in the routerservice and maybe create an object in CreateRecordRequest with an apply that take the APICreateRecordRequest and do the mapping in there.

Comment (IoannisLazaridis-oc) [reply app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala]: Done ✅ Included apply() method to RouterCreateRecordRequest and moved the mapping to RouterService to separate logic from controller

**Review by pauloncast — APPROVED** [review]

File: app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala:39]: can we move this to the CReateRecordController as this is the only place where we use it.

Comment (mauro-fanton) [reply app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala:39]: one sec hold that thought for ammoment

Comment (mauro-fanton) [reply app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala]: yes, I think this should be in the CreateRecordController as it is only use in there and is not generic

File: app/uk/gov/hmrc/tradergoodsprofiles/models/errors/ErrorResponse.scala
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofiles/models/errors/ErrorResponse.scala:102]: why this is commented out? Do you not need that?

File: test/uk/gov/hmrc/tradergoodsprofiles/connectors/RouteConnectorSpec.scala
Comment (mauro-fanton) [inline test/uk/gov/hmrc/tradergoodsprofiles/connectors/RouteConnectorSpec.scala:187]: dont need hc at the end, this is implicit

File: test/uk/gov/hmrc/tradergoodsprofiles/connectors/RouteConnectorSpec.scala
Comment (mauro-fanton) [inline test/uk/gov/hmrc/tradergoodsprofiles/connectors/RouteConnectorSpec.scala:198]: this is not TGP, should not be here. This may be my doing (copy and paste). Can this be removed please? I think there are several occurrences of this

File: app/uk/gov/hmrc/tradergoodsprofiles/controllers/RemoveRecordController.scala
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofiles/controllers/RemoveRecordController.scala:71]: It seem this is alway a 400 error. Can we rename maybe to BadRequestErrorResponse and refactor the object to inclide Code = BAD_REQUEST. Ping me if you want a chat about this

File: app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala]: this will be used by update endpoint eventually so its better to have it in validation support and may be rename to appropriate method as part of update endpoint

File: app/uk/gov/hmrc/tradergoodsprofiles/models/errors/ErrorResponse.scala
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofiles/models/errors/ErrorResponse.scala:102]: removed

File: app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala
Comment (mauro-fanton) [reply app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala]: Yes, ok, a more generic name would be fine. But again we are making the mistake of looking into the feature. The update record is already being done by Paul and Paul may uses different validation, so we may never have the time to refactor or change. So would have been better just to think about the create and then refactor when it is needed it.

Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofiles/utils/ValidationSupport.scala]: renamed the method name to be generic and Paul would be reusing this for update endpoint. thanks

File: app/uk/gov/hmrc/tradergoodsprofiles/models/responses/CreateRecordResponse.scala
Comment (pauloncast) [inline app/uk/gov/hmrc/tradergoodsprofiles/models/responses/CreateRecordResponse.scala:82]: I've just seen this. We don't need to validate response body, so this validate is not needed.
This file can be made to mimic this https://github.com/hmrc/trader-goods-profiles-router/blob/main/app/uk/gov/hmrc/tradergoodsprofilesrouter/models/response/CreateOrUpdateRecordResponse.scala

**Review by pauloncast — DISMISSED** [review]

**Review by pauloncast — APPROVED** [review]

**Pull request merged**
